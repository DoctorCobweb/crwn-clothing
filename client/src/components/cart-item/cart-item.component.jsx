import React from 'react';

import {
  CartItemContainer,
  CartItemImage,
  ItemDetailsContainer,
  NameContainer,
  PriceContainer 
} from './cart-item.styles.jsx';

export const CartItem = ({ item: { imageUrl, price, name, quantity } }) => (
  <CartItemContainer>
    <CartItemImage src={imageUrl} alt="item" />
    <ItemDetailsContainer>
      <NameContainer>{name}</NameContainer>
      <PriceContainer>${quantity} x {price}</PriceContainer>
    </ItemDetailsContainer>
  </CartItemContainer>
);

export default React.memo(CartItem);