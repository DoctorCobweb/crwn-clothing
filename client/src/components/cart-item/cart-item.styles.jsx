import styled from 'styled-components';

export const CartItemContainer = styled.div`
  width: 100%;
  display: flex;
  height: 80px;
  margin-bottom: 15px;
`;
CartItemContainer.displayName = 'CartItemContainer';


export const CartItemImage = styled.img`
  width: 30%
`; 
CartItemImage.displayName = 'CartItemImage';

export const ItemDetailsContainer = styled.div`
  width: 70%;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
  padding: 10px 20px
`;
ItemDetailsContainer.displayName = 'ItemDetailsContainer';

export const NameContainer = styled.span`
  font-size: 16px
`;
NameContainer.displayName = 'NameContainer';

export const PriceContainer = styled.span`
`;
PriceContainer.displayName = 'PriceContainer';