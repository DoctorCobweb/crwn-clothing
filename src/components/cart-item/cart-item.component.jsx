import React from 'react';

import {
  CartItemContainer,
  CartItemImage,
  ItemDetailsContainer,
  NameContainer,
  PriceContainer 
} from './cart-item.styles.jsx';
// import './cart-item.styles.scss';

const CartItem = ({ item: { imageUrl, price, name, quantity } }) => (
  <CartItemContainer>
    <CartItemImage src={imageUrl} alt="item" />
    <ItemDetailsContainer>
      <NameContainer>{name}</NameContainer>
      <PriceContainer>${quantity} x {price}</PriceContainer>
    </ItemDetailsContainer>
  </CartItemContainer>
);

// const CartItem = ({ item: { imageUrl, price, name, quantity } }) => (
//   <div className="cart-item">
//     <img src={imageUrl} alt="item" />
//     <div className="item-details">
//       <span className="name">{name}</span>
//       <span className="price">${quantity} x {price}</span>
//     </div>
//   </div>
// );

export default CartItem;