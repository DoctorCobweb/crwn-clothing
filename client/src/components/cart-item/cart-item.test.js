import React from 'react';
import { shallow } from 'enzyme';

import { CartItem } from './cart-item.component';

describe('CartItem component tests', () => {
  let wrapper;
  const mockImageUrl = 'https://example.com/image';
  const mockName = 'shorts';
  const mockQuantity = 2;
  const mockPrice = 11;

  beforeEach(()=> {
    const mockItem = {
      imageUrl: mockImageUrl,
      name: mockName,
      quantity: mockQuantity,
      price: mockPrice,
    }

    wrapper = shallow(<CartItem item={mockItem} />)
  })

  it('CartItem component should match snapshot', ()=> {
    expect(wrapper.debug()).toMatchSnapshot();
  })

  it('CartItemImage should have correct imageUrl set as src', ()=> {
    expect(wrapper.find('CartItemImage').prop('src')).toBe(mockImageUrl);
  })

  it('NameContainer should have name set as its text', () => {
    expect(wrapper.find('NameContainer').text()).toBe(mockName);
  })

  it('PriceContainer should correctly display text', ()=>{
    const priceText = `$${mockQuantity} x ${mockPrice}`;
    expect(wrapper.find('PriceContainer').text()).toBe(priceText);
  })

})