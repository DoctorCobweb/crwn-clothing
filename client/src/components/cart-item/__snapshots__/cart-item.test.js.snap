// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem component tests CartItem component should match snapshot 1`] = `
"<CartItemContainer>
  <CartItemImage src=\\"https://example.com/image\\" alt=\\"item\\" />
  <ItemDetailsContainer>
    <NameContainer>
      shorts
    </NameContainer>
    <PriceContainer>
      $
      2
       x 
      11
    </PriceContainer>
  </ItemDetailsContainer>
</CartItemContainer>"
`;
